### PR TITLE
refactor: clean up code

### PR DESCRIPTION
**Pull request by manniL**
* Use guarding ifs to return early (Hope that works out with Post/Prepatches 😲)
* Remove unnecessary `else` statements
* Use `LINQ` to simplify where possible
* Use speaking variables instead of comments where applicable
* Further simplifications

I'd be eager for feedback.

### CODE REVIEW

File: ExtraRoles/EndGamePatch.cs [hunk 1]
```diff
@@ -8,45 +9,37 @@ namespace ExtraRolesMod
     [HarmonyPatch(typeof(EndGameManager), "SetEverythingUp")]
     public static class EndGamePatch
     {
-        public static bool Prefix(EndGameManager __instance)
+        public static bool Prefix()
```
Comment (manniL) [inline ExtraRoles/EndGamePatch.cs:12]: Removed the argument here as it wasn't used.

File: ExtraRoles/PlayerTools.cs [hunk 2]
```diff
@@ -11,26 +11,20 @@ public static class PlayerTools
...
-                {
-                    isInfected = true;
                     break;
-                }
-                if (!isInfected)
-                {
-                    CrewmateIds.Add(player);
-                }
+                
+                CrewmateIds.Add(player);
```
Comment (manniL) [inline ExtraRoles/PlayerTools.cs:20]: Removed the flag here because the break happened anyway.

**Comment by NotHunter101** [comment]
Just so you don't waste your time, I will be pushing a major codebase rewrite soon. (meaning most of these commits won't be easily mergible)

**Comment by manniL** [comment]
> 
> 
> Just so you don't waste your time, I will be pushing a major codebase rewrite soon. (meaning most of these commits won't be easily mergeable)

Alright, thanks for the info! Will add a new PR then :)

File: ExtraRoles/SetInfectedPatch.cs [hunk 3]
```diff
@@ -33,20 +33,20 @@ public static void Postfix(Il2CppReferenceArray<GameData.PlayerInfo> JPGEIBIBJPJ
...
                 crewmates.RemoveAt(MedicRandom);
-                byte MedicId = MedicSettings.Medic.PlayerId;
+                var MedicId = MedicSettings.Medic.PlayerId;
 
                 writer.Write(MedicId);
                 AmongUsClient.Instance.FinishRpcImmediately(writer);
             }
 
-            if (crewmates.Count > 0 && (rng.Next(1, 101) <= HarmonyMain.officerSpawnChance.GetValue()))
+            if (crewmates.Count > 0 && rng.Next(1, 101) <= HarmonyMain.officerSpawnChance.GetValue())
```
Comment (manniL) [inline ExtraRoles/SetInfectedPatch.cs:42]: Ideally the spawning could be an own function taking role, value, etc.

**Pull request closed without merging**